temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1396 @@
+                job_sets[3].append(job)
+
+        # Apply every job set, running after each set
+        for job_set in job_sets:
+            if not job_set:
+                continue
+            for job in job_set:
+                # Let errors bubble out of these to the caller to fail the task
+                self._apply(job)
+
+            # Run one iteration of the loop
+            self._run_once()
+
+        # Return pending commands
+        return self._pending_commands
+
+    def _apply(
+        self, job: temporalio.bridge.proto.workflow_activation.WorkflowActivationJob
+    ) -> None:
+        if job.HasField("cancel_workflow"):
+            self._apply_cancel_workflow(job.cancel_workflow)
+        elif job.HasField("fire_timer"):
+            self._apply_fire_timer(job.fire_timer)
+        elif job.HasField("query_workflow"):
+            self._apply_query_workflow(job.query_workflow)
+        elif job.HasField("notify_has_patch"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("remove_from_cache"):
+            # Ignore, handled externally
+            pass
+        elif job.HasField("resolve_activity"):
+            self._apply_resolve_activity(job.resolve_activity)
+        elif job.HasField("resolve_child_workflow_execution"):
+            self._apply_resolve_child_workflow_execution(
+                job.resolve_child_workflow_execution
+            )
+        elif job.HasField("resolve_child_workflow_execution_start"):
+            self._apply_resolve_child_workflow_execution_start(
+                job.resolve_child_workflow_execution_start
+            )
+        elif job.HasField("resolve_request_cancel_external_workflow"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("resolve_signal_external_workflow"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("signal_workflow"):
+            self._apply_signal_workflow(job.signal_workflow)
+        elif job.HasField("start_workflow"):
+            self._apply_start_workflow(job.start_workflow)
+        elif job.HasField("update_random_seed"):
+            # TODO(cretz): This
+            pass
+        else:
+            raise RuntimeError(f"Unrecognized job: {job.WhichOneof('variant')}")
+
+    def _apply_cancel_workflow(
+        self, job: temporalio.bridge.proto.workflow_activation.CancelWorkflow
+    ) -> None:
+        # TODO(cretz): Details or cancel message or whatever?
+        if self._primary_task:
+            self._primary_task.cancel()
+
+    def _apply_fire_timer(
+        self, job: temporalio.bridge.proto.workflow_activation.FireTimer
+    ) -> None:
+        handle = self._pending_timers.pop(job.seq, None)
+        if not handle:
+            raise RuntimeError(f"Failed finding timer handle for sequence {job.seq}")
+        # Mark cancelled so the cancel() from things like asyncio.sleep() don't
+        # invoke _timer_handle_cancelled
+        handle._cancelled = True
```

Comment:
This is a bit weird, I'd rename to `_resolved` or something more meaningful.

Reply:
This is not my name: https://github.com/python/cpython/blob/v3.10.4/Lib/asyncio/events.py#L38

Reply:
I figured, might be worth checking in `_timer_handle_cancelled` instead but I haven't looked into this too much https://github.com/python/cpython/blob/9d38120e335357a3b294277fd5eff0a10e46e043/Lib/asyncio/events.py#L150

Reply:
Looks like the check in `_timer_handle_cancelled` is enough to ignore this timer so you don't even need to mark it in any meaningful way.

Reply:
Hrmm, yeah. There was some other reason I had to mark this cancelled, but maybe it was when I had more logic in `_timer_handle_cancelled`. I will try to remove setting this field and confirm behavior is not affected.

Reply:
Removed this assignment and confirmed everything is ok